singer/src/main/java/com/pinterest/singer/writer/kafka/CommittableKafkaWriter.java

```diff
@@ -138,20 +148,20 @@ public void startCommit() throws LogStreamWriterException {
       }
     }
 
-    commitableBuckets = new HashMap<>();
-    committableMapOfTrackedMessageMaps = new HashMap<>();
-    committableMapOfInvalidMessageMaps = new HashMap<>();
-    committableMapOfOriginalIndexWithinBucket = new HashMap<>();
-
+    committableBuckets.clear();
```

Comment:
In my past experience it's better to create a new data structure from GC cleanliness stand point.